relay_messages.go

```diff
@@ -94,14 +95,18 @@ func (cr lazyCallRes) OK() bool {
 type lazyCallReq struct {
 	*Frame
 
-	caller, method, delegate, key, as []byte
-
 	checksumTypeOffset             uint16
 	arg2StartOffset, arg2EndOffset uint16
 	arg3StartOffset                uint16
 
-	checksumType     ChecksumType
-	isArg2Fragmented bool
+	caller, method, delegate, key, as []byte
+	arg2Appends                       []relay.KeyVal
+	checksumType                      ChecksumType
+	isArg2Fragmented                  bool
+
+	// Intentionally an array to combine allocations with that of lazyCallReq
+	// TODO(echung): pool the appends buffers
```

Comment:
Just to be clear, this implies that arrays have a performance overhead we'd like to avoid in future -- is that the case?

I think the options are:

1. Don't use array backing because it's too high performance, and instead leave a TODO for pooling
2. Use an array backing, and no TODO for pooling since the array doesn't have a noticeable impact even on the "no append" case
3. Use an array backing since it improves performance, but still has a noticeable impact, so leave a TODO

The TODO implies (3), but I wanted to confirm whether the benchmark profiling shows enough of an impact on the no-append case